pkg/gateway/errors/errors.go

```diff
@@ -341,6 +342,11 @@ var Codes = errorCodeMap{
 		Description:    "The bucket lifecycle configuration does not exist",
 		HTTPStatusCode: http.StatusNotFound,
 	},
+	ErrNoSuchBucketPossibleAPIEndpoint: {
+		Code:           "ErrNoSuchBucketPossibleAPIEndpoint",
+		Description:    "Using lakeFS API URI as the endpoint. Try removing the 'api/v1/' part from the endpoint.",
```

Comment:
Perhaps afford people an explanation? I'm thinking of the poor user who has a repo named "api" and manages to mitype a branch name "v1"

Reply:
Hmmm... good thing they don't deduct bandwidth for errors from my paycheck.  I'd go with a definitive error message, followed by a suggestion:

```suggestion
		Description:    `Repository "api" not found; this can happen if your endpoint URL mistakenly ends in "` + apiutil.BaseURL + `".`,
```

(the `` ` `` is an alternative quotation mark that reduces the number of backslashes needed here.)